x/operator/types/genesis.go

```diff
@@ -0,0 +1,126 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func NewGenesisState(
+	operators []OperatorInfo,
+	records []OperatorConsKeyRecord,
+) *GenesisState {
+	return &GenesisState{
+		Operators:       operators,
+		OperatorRecords: records,
+	}
+}
+
+// DefaultGenesis returns the default genesis state
+func DefaultGenesis() *GenesisState {
+	return NewGenesisState([]OperatorInfo{}, []OperatorConsKeyRecord{})
+}
+
+// Validate performs basic genesis state validation returning an error upon any
+// failure.
+func (gs GenesisState) Validate() error {
+	operators := make(map[string]struct{}, len(gs.Operators))
+	for _, op := range gs.Operators {
+		address := op.EarningsAddr
+		if _, found := operators[address]; found {
+			return errorsmod.Wrapf(
+				ErrInvalidGenesisData,
+				"duplicate operator address %s", address,
+			)
+		}
+		_, err := sdk.AccAddressFromBech32(address)
+		if err != nil {
+			return errorsmod.Wrapf(
+				ErrInvalidGenesisData,
+				"invalid operator address %s: %s", address, err,
+			)
+		}
+		operators[address] = struct{}{}
+		if op.ClientChainEarningsAddr != nil {
+			lzIDs := make(map[uint64]struct{}, len(op.ClientChainEarningsAddr.EarningInfoList))
+			for _, info := range op.ClientChainEarningsAddr.EarningInfoList {
+				lzID := info.LzClientChainID
+				if _, found := lzIDs[lzID]; found {
+					return errorsmod.Wrapf(
+						ErrInvalidGenesisData,
+						"duplicate lz client chain id %d", lzID,
+					)
+				}
+				lzIDs[lzID] = struct{}{}
+				// TODO: consider removing this check for non-EVM chains
```

Comment:
To support non-EVM chains, we can introduce a new field or parameter that identifies the type of client chain and select different validation logic depending on the type of chain